.github/workflows/prbuild.yaml

```diff
@@ -1,13 +1,110 @@
 name: Build and Test Pull Request
 
 # Trigger the workflow on push or pull request
-on: [push, pull_request]
+on: 
+  push:
+    branches-ignore:
+      - "main"
+  # Trigger this workflow when the PR is labeled or unlabelled
+  # as well as opened or updated.
+  pull_request:
+    types: [opened, labeled, unlabeled, synchronize]
 
 env:
   GOPROXY: https://proxy.golang.org/
   SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
 jobs:
+  check-label:
+    name: Check release-note label set
+    runs-on: ubuntu-latest
+    steps:
+      - uses: mheap/github-action-required-labels@v1
+        with:
+          mode: exactly
+          count: 1
+          labels: "release-note/major, release-note/minor, release-note/small, release-note/docs, release-note/infra, release-note/none-required"
+  check-changelog:
+    name: Check for changelog file
+    needs:
+      - check-label
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - run: go run ./hack/actions/check-changefile-exists.go
+      env:
+        PR_NUMBER: ${{ github.event.number }}
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Related to previous comment, I'm thinking it might be easier to move these jobs into a separate workflow that can be triggered just for PRs, on all the label/unlabel/etc events, and leave `prbuild.yaml` more or less as-is.